pandas/core/reshape/reshape.py

```diff
@@ -921,7 +922,9 @@ def check_len(item, name):
                 dtype=dtype,
             )
             with_dummies.append(dummy)
-        result = concat(with_dummies, axis=1)
+        concatted = concat(with_dummies, axis=1)
+        assert isinstance(concatted, DataFrame)
+        result = concatted
```

Comment:
This is the same annoying mypy issue as in #30565. don't know if there's a better way to do this